scrape/manager_test.go

```diff
@@ -720,34 +720,88 @@ scrape_configs:
 func TestManagerCTZeroIngestion(t *testing.T) {
 	const mName = "expected_counter"
 
+	type exp struct {
+		value float64
+		ts    int64
+	}
+
 	for _, tc := range []struct {
 		name                  string
-		counterSample         *dto.Counter
+		counterSampleProto    *dto.Counter
+		counterSampleText     string
 		enableCTZeroIngestion bool
+		exp                   []exp
+		typ                   string
 	}{
 		{
-			name: "disabled with CT on counter",
-			counterSample: &dto.Counter{
+			name: "Protobuf disabled with CT on counter",
+			counterSampleProto: &dto.Counter{
 				Value: proto.Float64(1.0),
 				// Timestamp does not matter as long as it exists in this test.
 				CreatedTimestamp: timestamppb.Now(),
 			},
+			typ: "application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited",
 		},
 		{
-			name: "enabled with CT on counter",
-			counterSample: &dto.Counter{
+			name: "Protobuf enabled with CT on counter",
+			counterSampleProto: &dto.Counter{
 				Value: proto.Float64(1.0),
 				// Timestamp does not matter as long as it exists in this test.
 				CreatedTimestamp: timestamppb.Now(),
 			},
 			enableCTZeroIngestion: true,
+			typ:                   "application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited",
 		},
 		{
-			name: "enabled without CT on counter",
-			counterSample: &dto.Counter{
+			name: "Protobuf enabled without CT on counter",
+			counterSampleProto: &dto.Counter{
 				Value: proto.Float64(1.0),
 			},
 			enableCTZeroIngestion: true,
+			typ:                   "application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited",
+		},
+		{
+			name: "OMText disabled with CT on counter",
+			counterSampleText: `# TYPE expected_counter counter
+expected_counter 17.0 1520879607.789
+expected_counter_created 1000
+# EOF`,
+			exp: []exp{{
+				value: 17.0,
+				ts:    1520879607789,
+			}},
```

Comment:
Not sure if we want to verify this as well, since it will need a refactoring of the whole test, so asking first: Do we want to verify that `expected_counter_created` is parsed as a separate series when EnableCTZeroIngestion is false?

Reply:
the rationale here is that when ct ingestion is disabled, and we get a created line anyway, it doesn't add a new entry (is it called series in this case). I was just going off of the protobuf tests. I do think its valuable as we wouldn't want a sample appended if created is exported but not enabled. 

Lemme know if theres any gaps in my logic here

Reply:
> protobufs have some niceities like .GetValue() and .IsValid() that isn't quite possible for the text format unless I explicitly add the exp struct. The tests at the bottom would look kinda weird with the switch case

Also this to consider as to why i have that exp struct

Reply:
i've now added a test case for if _created create new metrics when ct ingestion is disabled